### ISSUE TITLE
Schema evaluation failed when use relative path in $ref

### ISSUE THREAD

**Issue by kool79** [harrel56/json-schema#63]
schema 1:
```
{
  "$schema":    "https://json-schema.org/draft/2020-12/schema",
  "$id":             "data/schemas/sample-schema.json",
  "type":       "object",
  "properties": {"xx": {"$ref": "common.json#/$defs/reqFields"}}
}
```
schema with common definitions
```
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "$id":     "data/schemas/common.json",
  
"$defs":   {
    "reqFields": {"required": ["yyy"]}
  }
}
```
Both files are located in /data/schemas subdirectory under current dir. 
(The IntelliJ IDEA perfectly navigates to definition in common.json file when I ctrl+click to the 'reqFields' pointer)
I use the next resolver to handle local paths:
```
SchemaResolver resolver = (String uri) -> {
    if (uri.contains(":")){  // skip all links with :
        return SchemaResolver.Result.empty();
    try{
       var file = new File(uri);   // locate file by relative link in 'uri'
       return SchemaResolver.Result.fromProviderNode(objectMapper.readTree(file));
   }catch (IOException e){
	throw new RuntimeException(e, "Cannot load schema file [%s]".formatted(uri));
}};
```
My resolver is called 2 times:
 1st time for `uri = data/schemas/common.json#/$defs/reqFields` (which is valid and already resolved)
  2nd time for `uri= data/schemas/data/schemas/common.json#/$defs/reqFields` (which is obviously invalid)
  
  The second call is from the code dev.harrel.jsonschema.EvaluationContext#resolveSchema, last line:
  ``` java 
       private Optional<Schema> resolveSchema(String ref) {
        String resolvedUri = UriUtil.resolveUri(dynamicScope.element(), ref);
        return OptionalUtil.firstPresent(
                () -> Optional.ofNullable(schemaRegistry.get(resolvedUri)),
                () -> Optional.ofNullable(schemaRegistry.getDynamic(resolvedUri)),
                () -> resolveExternalSchema(resolvedUri)
        );
    }
```
The input parameter ('uri')  already has the value (`data/schemas/common.json#/$defs/reqFields`) but the `resolvedUri` is evaluated from:
```
  resolveUri = UriUtil.resolveUri("data/schemas/sample-schema.json", "data/schemas/common.json#/$defs/reqFields")
```
as incorrect value `"data/schemas/data/schemas/common.json#/$defs/reqFields"`.
Note that schemaRegistry already contains the schema for `data/schemas/common.json` (it was loaded calling method (methos which called current one) `dev.harrel.jsonschema.EvaluationContext#resolveExternalSchema`

**Comment by harrel56** [comment]
Thanks for the feedback! Gonna look into that later today

**Comment by harrel56** [comment]
Alright, this was rather easy to fix - already released the fix with version 1.3.2 (visibility in maven central might take some time). Feel free to reopen if something is still off.
